validator/app/src/compute_horde_validator/validator/allowance/utils/supertensor.py

```diff
@@ -269,9 +337,40 @@ def get_current_block(self) -> int:
         return current_block - 5
 
     def close(self):
-        self.loop.run_until_complete(self.bittensor.close())
-        if self.archive_bittensor is not None:
-            self.loop.run_until_complete(self.archive_bittensor.close())
+        if self._closed or self.loop is None or self._background_thread is None:
+            return
+
+        self._closed = True
+
+        async def _close_resources():
+            await self.bittensor.close()
+            if self.archive_bittensor is not None:
+                await self.archive_bittensor.close()
+
+        future = asyncio.run_coroutine_threadsafe(_close_resources(), self.loop)
+
+        try:
+            future.result(timeout=DEFAULT_TIMEOUT)
+        except (TimeoutError, concurrent.futures.TimeoutError):
```

Comment:
```suggestion
        except TimeoutError:
```

AFAIK they are the same.